emulator.c

```diff
@@ -81,8 +92,18 @@ execute_instruction(i8080 *cpu, uint8_t opcode)
         break;
       }
     case 0x19: // NOLINT
-      printf("DAD D");
-      break;
+      {        // DAD D
+        uint16_t sum
+            = (cpu->h << 8) // NOLINT
+              | (cpu->l);   // convert reg pair h,l to 16 bit int // NOLINT
+        sum = sum + cpu->d;
+
+        cpu->h = sum >> 8;                   // NOLINT
+        cpu->l = sum & 255;                  // NOLINT
+        update_carry_flag(cpu, sum > 65535); // NOLINT
```

Comment:
Minor style thing but I think it's clearer to write things in hex. sum > 0xFFFF - easier to be certain you mean the largest 16-bit number.